molgenis-data/src/main/java/org/molgenis/data/support/DefaultEntity.java

```diff
@@ -275,7 +276,15 @@ public Entity getEntity(String attributeName)
 		EntityMetaData ref = attribute.getRefEntity();
 		ids = ids.stream().map(attribute.getDataType()::convert).collect(Collectors.toList());
 
-		return dataService.findAll(ref.getName(), IN(ref.getIdAttribute().getName(), ids));
+		String entityIdName = ref.getIdAttribute().getName();
+		Map<Object, Entity> entities = new HashMap<Object, Entity>();
```

Comment:
How about we stream it like this then, preserving order, by turning it into separate lookups?
I'm a bit torn here in the tradeoff between order preservation, memory footprint, query count, and clean code.

The other option, I'd think, is that in MySQL, we could also override the getEntities() method of the DefaultEntity, which would then join the MREF table with the refEntity table, sorting in order, and stream the results efficiently from that last-minute single select.